### DISCUSSION TITLE
Updates to actor properties are not reflected in Jupyter notebook Plotter

### DISCUSSION THREAD

**Discussion by ScottH** [pyvista/pyvista#2002] [Q&A]
I'm trying to interactively modify actor properties in a Jupyter notebook and to have the changes appear in a Plotter.

E.g. this code in a Notebook cell displays a sphere in an interactive Plotter in my notebook:
```
import pyvista as pv

mesh = pv.Sphere()

p = pv.Plotter(notebook=True)
actor = p.add_mesh(mesh)

#p.add_checkbox_button_widget(toggle_vis, value=True)

p.show(jupyter_backend='pythreejs') # updates to actor are not reflected in Plotter
#p.show() # updates to actor appear after I click in Plotter
```

If I call this in the next cell the change to the actor does not happen with the pythreejs backend but does appear in the Plotter with the default backend once I click in the cell. 

```
actor.SetVisibility(False)
```

My goal is to set up a scene in a pythreejs Plotter in a notebook and then have changes to the actors made in subsequent cells be reflected in the Plotter display. Is this possible? This would enable simple animations and dynamic displays connected to ipywidgets. Is there a function I can call on the Plotter object to force a redraw after I update the actor properties? This is similar to the ask in #1950, but inside a Jupyter notebook.

Thanks!

**Comment by banesullivan** [comment]
Only the `ipyvtklink` notebook back-end will be compatible with this sort of interactiveity/dynamicness. The rest of the backends (pythreejs) export a static state of the scene and embed that in the notebook (no way to update without re-embedding or creating a whole new interactive widget that interfaces to PyVista). 

`ipyvtklink` provides a dynamic link that you can update in this manner.

**Reply by ScottH** [reply]
Thanks Bane,
ipyvtklink seems to behave the same way as the demo code above: I have to click rotate the plotter to get the update to happen on the screen. How do I trigger a re-render after changing the actor properties?

-Scott

**Reply by akaszynski** [reply]
It's not perfect, but this will get you started:
```py
from ipywidgets import Checkbox, VBox
import pyvista as pv

mesh = pv.Sphere()

p = pv.Plotter(notebook=True)
actor = p.add_mesh(mesh)

ipyvtk_widget = p.show(jupyter_backend='ipyvtklink', return_viewer=True) # updates to actor are not reflected in Plotter

cbox = Checkbox(
    value=True,
    description='Visible',
    disabled=False,
    indent=False,
)

def change_vis(value):
    actor.SetVisibility(value['owner'].value)
    # force render
    ipyvtk_widget.quick_render()

cbox.observe(change_vis)

VBox(children=[cbox, ipyvtk_widget])
```

https://user-images.githubusercontent.com/11981631/149419038-2816b340-766f-4bb6-bdad-6dbc38dfc97a.mp4
